src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -183,6 +227,47 @@ DecisionTree<FitnessFunction,
       minimumLeafSize, minimumGainSplit, maximumDepth, dimensionSelector);
 }
 
+//! Construct and train with weights.
+template<typename FitnessFunction,
+        template<typename> class NumericSplitType,
+        template<typename> class CategoricalSplitType,
+        typename DimensionSelectionType,
+        typename ElemType,
+        bool NoRecursion>
+template<typename MatType, typename LabelsType, typename WeightsType>
+DecisionTree<FitnessFunction,
+        NumericSplitType,
+        CategoricalSplitType,
+        DimensionSelectionType,
+        ElemType,
+        NoRecursion>::DecisionTree(const DecisionTree& other,
+                                   MatType data,
+                                   LabelsType labels,
+                                   const size_t numClasses,
+                                   WeightsType weights,
+                                   const size_t minimumLeafSize,
+                                   const double minimumGainSplit,
+                                   const std::enable_if_t<
+                                           arma::is_arma_type<
+                                                   typename std::remove_reference<
+                                                           WeightsType>::type>::value>*):
+        NumericAuxiliarySplitInfo(other),
+        CategoricalAuxiliarySplitInfo(other)  // other info does need to copy
```

Comment:
Just like `src/mlpack/methods/decision_tree/decision_tree_impl.hpp`, I would recommend to change this block like that
```
template<typename FitnessFunction,
         template<typename> class NumericSplitType,
         template<typename> class CategoricalSplitType,
         typename DimensionSelectionType,
         typename ElemType,
         bool NoRecursion>
template<typename MatType, typename LabelsType, typename WeightsType>
DecisionTree<FitnessFunction,
             NumericSplitType,
             CategoricalSplitType,
             DimensionSelectionType,
             ElemType,
             NoRecursion>::DecisionTree(
    const DecisionTree& other,
    MatType data,
    LabelsType labels,
    const size_t numClasses,
    WeightsType weights,
    const size_t minimumLeafSize,
    const double minimumGainSplit,
    const std::enable_if_t<
        arma::is_arma_type<
        typename std::remove_reference<
        WeightsType>::type>::value>*) :
    NumericAuxiliarySplitInfo(other),
    CategoricalAuxiliarySplitInfo(other)  // other info does need to copy
```